src/controller/CourseController.ts

```diff
@@ -47,9 +47,21 @@ export class CourseController {
     async filter(request: Request, response: Response, next?: NextFunction) {
         try {
             const parsedParams = await this.parseQuery(request.query);
-            const courses = await this.courseRepository.find({
-                where: parsedParams
-            });
+            let start: number = request.query["_start"];
+            let end: number = request.query["_end"];
+            const order = request.query["_order"];
+            const sort = request.query["_sort"];
+            const courseName = request.query["name_like"]
```

Comment:
interestingly, i saw that the `GET` request contained this additional query param `name_like` used for the search bar on the course page, so this new stuff i added covers that and it seems to work pretty well.

by default, the sql query will just grab all the records and not filter if `name_like` isn't there and the other few query params also seems to have default values sent from the frontend (e.x. `order` will be ASC by on default)

Reply:
oops, just saw this issue https://github.com/sandboxnu/pharmd-tracker/issues/44. i will create a separate pr that branches off of this one for searching courses by the subject/name/number. this only covers name 😓

Reply:
made this pr to cover all parts of the course search https://github.com/sandboxnu/pharmd-tracker-backend/pull/48